web/libs/ui/src/lib/select/select.tsx

```diff
@@ -49,7 +50,9 @@ export const Select = forwardRef(
     const [query, setQuery] = useState<string>("");
     const valueRef = useRef<any>();
     let initialValue = defaultValue?.value ?? defaultValue ?? externalValue?.value ?? externalValue;
-
+    if (selectFirstIfEmpty && initialValue !== null || initialValue !== undefined) {
```

Comment:
```suggestion
    if (selectFirstIfEmpty && initialValue === null || initialValue === undefined) {
```
Sorry I messed up the logic originally here